### DISCUSSION TITLE
Admin Password Request Email not Sending

### DISCUSSION THREAD

**Discussion by anohene1** [pocketbase/pocketbase#2496] [Q&A] [answered]
I've forgotten my admin password as it's been long since I last touched my project. I clicked on Forgot Password and entered my email address. I got a feedback that I should check my email, but it's been a very long time now and it's still not sending.

Edit: I'm using Windows 11. Pocketbase version 0.12.2

**Comment by ganigeorgiev** [comment, marked as answer]
PocketBase by default uses the `sendmail` unix command for sending email and it may not be reliable.

If you want to reset your admin password, you can manually update the db entry with `123456` for example with the `sqlite3` cli tool:

```sh
sqlite3 pb_data/data.db 'UPDATE _admins SET passwordHash="$2a$10$xWbkSks7OiTmrri2DU6QSe6OAekcdqkOrkcI1wi4uiv9SU2HTlHoq" where email="test@example.com"'
```
(_remove the email constraint if you don't remember your email in order to reset the password for all admins_)

With the upcoming v0.16.0 release there will be a new `admin` command that will make the process slightly easier:

```sh
./pocketbase admin update test@example.com 1234567890
```

**Reply by anohene1** [reply]
Thanks this didn't work exactly, but when I swapped the single quotes and double qoutes it worked. Like so: 
`sqlite3 pb_data/data.db "UPDATE _admins SET passwordHash='$2a$10$xWbkSks7OiTmrri2DU6QSe6OAekcdqkOrkcI1wi4uiv9SU2HTlHoq' where email='test@example.com'"`

**Reply by ThomasGysemans** [reply]
but what is the hash algorithm used for passwords?

**Reply by ganigeorgiev** [reply]
@ThomasGysemans We use bcrypt. In any case please don't hijack old discussions and if you have other question open a new one with more details about your use case and what you are trying to do.
